dbms/src/Storages/DeltaMerge/SegmentReadTask.cpp

```diff
@@ -542,10 +546,7 @@ static void checkPageID(
 
 void SegmentReadTask::doFetchPages(const disaggregated::FetchDisaggPagesRequest & request)
 {
-    // No page and memtable need to be fetched.
-    if (request.page_ids_size() == 0 && !needFetchMemTableSet())
-        return;
-
+    // No matter all delta data is cached or not, call FetchDisaggPages to release snapshot in WN.
```

Comment:
@JinheLin @breezewish 
What about adding another unary RPC method "CancelDisaggTask". When the MPP request on CN is finished or canceled, CN calls "CancelDisaggTask" to let the WN release the whole snapshot. Instead of sending "fetchPages" segment by segment?

This can
* decouple the optimization of 'fetchPages' and release WN snapshot
* reduce the total RPC calls between CN and RN
* handle the situation when the request is canceled in the CN

Reply:
@JinheLin Thanks, it looks good!

Reply:
Get it